### PR TITLE
Add support for building WASM variants alongside single-threaded to the build system

### PR DESCRIPTION
**Pull request by directhex**
This is a first cut at allowing single thread, multi thread, and perftrace wasm builds to all happen as a single "thing"

- [x] Add mono.wasmvariants subset to build variants
- [x] Add variants to WASM package, or their own packages, TBD
- [x] Add YAML to actually build this in CI

### CODE REVIEW

**Comment by directhex** [comment]
Right now, this is dumping the runtime variants in `artifacts/bin/mono/Browser.Wasm.Release/wasm/{multithread,perftrace}` and the native parts in `artifacts/bin/native/net7.0-Browser-Release-wasm/wasm/{multithread,perftrace}`

**Comment by directhex** [comment]
@steveisok I've eventually got this running with some very bad yaml. I tried parameterizing on runtimeVariant, but couldn't find a way to use the parameter in nameSuffix (which must be unique) without error. Moving from the old properties to the common MonoWasmBuildVariant property should make future parameterization easier.

**Comment by steveisok** [comment]
> @steveisok I've eventually got this running with some very bad yaml. I tried parameterizing on runtimeVariant, but couldn't find a way to use the parameter in nameSuffix (which must be unique) without error. Moving from the old properties to the common MonoWasmBuildVariant property should make future parameterization easier.

I'm ok with a follow up to make things better.

File: src/mono/mono.proj [hunk 1]
```diff
@@ -53,8 +53,9 @@
     <_CompilerTargetArch Condition="'$(RealTargetArchitecture)' != ''">$(RealTargetArchitecture)</_CompilerTargetArch>
     <RepositoryEngineeringCommonDir>$([MSBuild]::NormalizeDirectory('$(RepositoryEngineeringDir)', 'common'))</RepositoryEngineeringCommonDir>
     <CrossToolchainFile>$([MSBuild]::NormalizePath('$(RepositoryEngineeringCommonDir)', 'cross', 'toolchain.cmake'))</CrossToolchainFile>
-    <MonoWasmThreads Condition="'$(WasmEnableThreads)' == 'true' or '$(WasmEnablePerfTracing)' == 'true'">true</MonoWasmThreads>
-    <MonoWasmThreadsNoUser Condition="'$(WasmEnableThreads)' != 'true' and '$(WasmEnablePerfTracing)' == 'true'">true</MonoWasmThreadsNoUser>
+    <MonoWasmThreads Condition="'$(MonoWasmBuildVariant)' == 'singlethread'">false</MonoWasmThreads>
```
[the remark this answers is not in the document]
Comment (directhex) [reply src/mono/mono.proj:56]: To a degree that's intentional - the packaging logic renames the package when MonoWasmBuildVariant is non-null, so the default case of "isn't set" results in the old single threaded package under the un-adorned name

**Review by lewing — APPROVED** [review]

**Pull request merged**